### PR TITLE
ci: update buildkite's hook

### PR DESCRIPTION
**Pull request by yihau**
#### Summary of Changes

- remove unnecessary files
- unify the header

### CODE REVIEW

**Comment by mvines** [comment]
The reason why those .sh symlinks exist is to get shellcheck coverage, as shellcheck won't check shell files without an extension

**Comment by t-nelson** [comment]
> The reason why those .sh symlinks exist is to get shellcheck coverage, as shellcheck won't check shell files without an extension

maybe we move the logic into real .sh, call that from the hook file with a comment to keep all the logic in the .sh? less deep lore to keep around

File: ci/shellcheck.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@ set -e
 cd "$(dirname "$0")/.."
 (
   set -x
-  git ls-files -- '*.sh' ':(exclude)ci/semver_bash' \
```
Comment (yihau) [inline ci/shellcheck.sh:9]: I think this one works.

I tried to make ci unhappy in https://github.com/solana-labs/solana/pull/31298/commits/a0fdfb40db8aa226a2669e474a06fb0f45f8c5c4 and the pipeline failed expectedly
Screenshot 2023-04-23 at 2 08 44 PM
(https://buildkite.com/solana-labs/solana/builds/94353#0187acb6-8f1e-4a3b-bf20-71924ad2a645)

File: ci/shellcheck.sh [hunk 2]
```diff
@@ -6,7 +6,7 @@ set -e
...
+  git ls-files -- '.buildkite/hooks' '*.sh' ':(exclude)ci/semver_bash' \
```
Comment (yihau) [inline ci/shellcheck.sh:9]: I think this one works.

I tried to make ci unhappy in https://github.com/solana-labs/solana/pull/31298/commits/a0fdfb40db8aa226a2669e474a06fb0f45f8c5c4 and the pipeline failed expectedly
Screenshot 2023-04-23 at 2 08 44 PM
(https://buildkite.com/solana-labs/solana/builds/94353#0187acb6-8f1e-4a3b-bf20-71924ad2a645)

Comment (mvines) [reply ci/shellcheck.sh:9]: Ah, that looks great then. Bye bye .sh

**Comment by yihau** [comment]
could I have an approve to ship this one 👻

**Review by mvines — APPROVED** [review]
sry, done

**Pull request merged**